python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -745,30 +745,42 @@ def parents_will_be_reconciled(
                 continue
 
             if not (
-                (
-                    parent in conditions_by_asset_partition
-                    and all(
-                        condition.decision_type == AutoMaterializeDecisionType.MATERIALIZE
-                        for condition in conditions_by_asset_partition[parent]
-                    )
+                parent in conditions_by_asset_partition
+                and all(
+                    condition.decision_type == AutoMaterializeDecisionType.MATERIALIZE
+                    for condition in conditions_by_asset_partition[parent]
                 )
-                # if they don't have the same partitioning, then we can't launch a run that
-                # targets both, so we need to wait until the parent is reconciled before
-                # launching a run for the child
-                and asset_graph.have_same_partitioning(parent.asset_key, candidate.asset_key)
+            ):
+                return False, ParentOutdatedAutoMaterializeCondition(
+                    parent_asset_key=parent.asset_key, parent_will_materialize=False
+                )
+
+            # if they don't have the same partitioning, then we can't launch a run that
+            # targets both, so we need to wait until the parent is reconciled before
+            # launching a run for the child
+            if not (
+                asset_graph.have_same_partitioning(parent.asset_key, candidate.asset_key)
                 and parent.partition_key == candidate.partition_key
             ):
-                return False
+                return False, ParentOutdatedAutoMaterializeCondition(
+                    parent_asset_key=parent.asset_key,
+                    parent_will_materialize=True,
+                    different_partitions=True,
+                )
 
             if isinstance(asset_graph, ExternalAssetGraph):
                 # if the parent is in a different repository, we can't launch a run that targets both,
                 # so we need to wait
                 if asset_graph.get_repository_handle(
                     candidate.asset_key
                 ) is not asset_graph.get_repository_handle(parent.asset_key):
-                    return False
+                    return False, ParentOutdatedAutoMaterializeCondition(
+                        parent_asset_key=parent.asset_key,
+                        parent_will_materialize=True,
+                        different_repositories=True,
+                    )
 
-        return True
+        return True, None
```

Comment:
what if we just had this return the AutoMaterializeCondition? then we could have a separate function to determine if that should be treated as a True or False